modules/programs/vim.nix

```diff
@@ -55,6 +55,24 @@ let
     in
       optionalString (value != null) ("set " + v);
 
+  plugins = flip map cfg.plugins (plugin:
+    if builtins.isString plugin
+    then warn
```

Comment:
Instead of `warn` and `throw` I would suggest using the `warnings` and `assertions` NixOS options as [described in the manual](https://nixos.org/nixos/manual/index.html#sec-assertions).

A single warning like "Specifying VIM plugins using strings is deprecated, found "plugin1", "plugin2" as strings" looks nicer and it also lets HM present the warnings in a more controlled manner. Same holds for the errors.

In this code you can either filter out any plugin that's not in `pkgs.vimPlugins` or simply assume that they all are there.

Reply:
It might also be nice to add a news entry conditional on `programs.vim.enable` that includes a description of this change.

Reply:
Sadly the warnings listed in `warnings` are not shown to the user. Maybe this is a bug in the nixos module implementation, but there is no warning in neither stdout of `nixos-rebuild` nor in the systemd user services for the activation..

Reply:
I will report it as separate issue